alembic/versions/20230706_04bbd03bf9f1_migrate_library_key_pair.py

```diff
@@ -0,0 +1,69 @@
+"""Migrate library key pair
+
+Revision ID: 04bbd03bf9f1
+Revises: f08f9c6bded6
+Create Date: 2023-07-06 14:40:17.970603+00:00
+
+"""
+import json
+import logging
+
+import sqlalchemy as sa
+from Crypto.PublicKey import RSA
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "04bbd03bf9f1"
+down_revision = "f08f9c6bded6"
+branch_labels = None
+depends_on = None
+
+log = logging.getLogger(f"palace.migration.{revision}")
+log.setLevel(logging.INFO)
+log.disabled = False
+
+
+def upgrade() -> None:
+    # Add the new columns as nullable, add the values, then make them non-nullable
+    op.add_column(
+        "libraries",
+        sa.Column("public_key", sa.Unicode(), nullable=True),
+    )
+    op.add_column(
+        "libraries",
+        sa.Column("private_key", sa.LargeBinary(), nullable=True),
+    )
+
+    # Now we update the value stored for the key pair
+    connection = op.get_bind()
+    libraries = connection.execute("select id, short_name from libraries")
+    for library in libraries:
+        setting = connection.execute(
+            "select cs.value from configurationsettings cs "
+            "where cs.library_id = (%s) and cs.key = 'key-pair'",
```

Comment:
For a per-library setting, the `where` clause should include a null external integration:
```
"where cs.library_id = (%s) and cs.external_integration_id IS NULL and cs.key = 'key-pair'",
```